### ISSUE TITLE
Installer architecture changes from x64 to x86 resulting in duplicate installer manifest error

### ISSUE THREAD

**Issue by mivsvit** [microsoft/winget-create#329]
### Brief description of your issue

A large number of existing manifests have duplicate installer entries (often the same installer for both x86 and x64 architectures). 

For example:
https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/WindowsTerminal/Preview/1.16.2641.0/Microsoft.WindowsTerminal.Preview.installer.yaml
https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Mattermost/MattermostDesktop/5.2.1/Mattermost.MattermostDesktop.installer.yaml
https://github.com/microsoft/winget-pkgs/blob/master/manifests/b/Bitwarden/Bitwarden/2022.11.0/Bitwarden.Bitwarden.installer.yaml

It is not possible to use `wingetcreate update` as the application returns the error `Manifest Error: Duplicate installer entry found.`

As a workaround an alternative manifest creation utility needs to be used such as `YamlCreate.ps1`

It is not possible to submit a different number of installers than are already present. 

### Steps to reproduce

Attempt to update the Mattermost package from 5.2.1 to 5.2.2 with 
`wingetcreate update  Mattermost.MattermostDesktop  -v 5.2.2  -s --interactive`

Submit the four URLs using the `InstallerUrl` fields below.

```
Installers:
- InstallerLocale: en-US
  Architecture: x64
  InstallerType: wix
  Scope: machine
  InstallerUrl: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x64.msi
  InstallerSha256: 41CED5DEF3FA881D46ED552C43F40C66FE33773FC980AE9807B85552FA74BB65
  ProductCode: '{77D38236-B4AD-4C27-88BE-098ED8ECBF91}'
- InstallerLocale: en-US
  Architecture: x86
  InstallerType: wix
  Scope: machine
  InstallerUrl: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x86.msi
  InstallerSha256: 1C1C11CCDC42687D07C7F5AC3565566FB257D9C21481257FEE63741A267D5BA1
  ProductCode: '{5C2B3508-5872-4BBB-B61D-E1AD65064E6C}'
- Architecture: x86
  InstallerType: nullsoft
  Scope: user
  InstallerUrl: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe
  InstallerSha256: 8C04D7CFB0BDDE12956F8B140E4DA6B7438167EDF616A7859C308E51C108BBD1
- Architecture: x86
  InstallerType: nullsoft
  Scope: user
  InstallerUrl: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe
  InstallerSha256: 8C04D7CFB0BDDE12956F8B140E4DA6B7438167EDF616A7859C308E51C108BBD1
  ```

### Expected behavior

Expect that non-unique InstallerUrls are not regarded as an error. Manifest validation succeeds. 

### Actual behavior

```
Manifest validation succeeded: False
Manifest Error: Duplicate installer entry found.
```

### Environment

```shell
Windows Package Manager v1.3.2691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.1219
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2691.0
```

**Comment by mivsvit** [comment]
That's correct - they are duplicate in the current manifest in winget-pkgs.

I.e. the URL at
https://github.com/microsoft/winget-pkgs/blob/16f1caec857476589e8aa9bfd837ae4486599e95/manifests/m/Mattermost/MattermostDesktop/5.2.1/Mattermost.MattermostDesktop.installer.yaml#L30
is the same as at
https://github.com/microsoft/winget-pkgs/blob/16f1caec857476589e8aa9bfd837ae4486599e95/manifests/m/Mattermost/MattermostDesktop/5.2.1/Mattermost.MattermostDesktop.installer.yaml#L35

My point is that it does not appear that this should be regarded as an error as there are many manifests like this.

**Comment by mivsvit** [comment]
OK I see what the issue is.
When submitting  each installer one by one the architecture is correct (3rd is `x64`, 4th is `x86`):

```
Updating 1 of 4 installers:
Architecture: x64
InstallerType: wix
Scope: machine
InstallerUrl: https://releases.mattermost.com/desktop/5.2.1/mattermost-desktop-5.2.1-x64.msi
InstallerSha256: a07239b5215d7177decff85408ece440c58189d371355bbc9eebddc18f214251
ProductCode: '{1F7E3DB0-37E1-4548-81D5-15FBFBBBFEEA}'

 What is the new installer url?: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x64.msi

Updating 2 of 4 installers:
Architecture: x86
InstallerType: wix
Scope: machine
InstallerUrl: https://releases.mattermost.com/desktop/5.2.1/mattermost-desktop-5.2.1-x86.msi
InstallerSha256: 9ce12955f7ef6c910fbb98d2ccc242864d69b352a82c419a9d4bb14c5241f6ab
ProductCode: '{D7FB69C0-9D40-48AB-8DAE-EFB2C8941EB9}'

 What is the new installer url?: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x86.msi

Updating 3 of 4 installers:
Architecture: x64
InstallerType: inno
Scope: user
InstallerUrl: https://releases.mattermost.com/desktop/5.2.1/mattermost-desktop-setup-5.2.1-win.exe
InstallerSha256: 0886e1155c4e4b5593fa18a0562e74224aee613d7be51f6e845bf05c38bb3937

 What is the new installer url?: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe

Updating 4 of 4 installers:
Architecture: x86
InstallerType: inno
Scope: user
InstallerUrl: https://releases.mattermost.com/desktop/5.2.1/mattermost-desktop-setup-5.2.1-win.exe
InstallerSha256: 0886e1155c4e4b5593fa18a0562e74224aee613d7be51f6e845bf05c38bb3937

 What is the new installer url?: https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe
 ```
 Wingetcreate then generates a `Duplicate installer entry` error and when displaying the manifest, the 3rd is incorrectly listed as `x86` not `x64` as submitted:
 
 ```
 Manifest validation succeeded: False
Manifest Error: Duplicate installer entry found.

 Would you like to discard this update and start over?: (y/n) n

 Would you like to edit your manifests?: (y/n) y

Manifest validation succeeded: False
Manifest Error: Duplicate installer entry found.

 Which manifest do you want to edit?:
  VERSION: Mattermost.MattermostDesktop.yaml
> INSTALLER: Mattermost.MattermostDesktop.installer.yaml
  DEFAULTLOCALE: Mattermost.MattermostDesktop.locale.en-US.yaml
  DONE
  
 Which manifest do you want to edit?:
  VERSION: Mattermost.MattermostDesktop.yaml
> INSTALLER: Mattermost.MattermostDesktop.installer.yaml
  DEFAULTLOCALE: Mattermost.MattermostDesktop.locale.en-US.yaml
  DONE

Filter menu items by typing a search query below.
 Which property would you like to edit?:
  InstallerLocale
  MinimumOSVersion
  Scope
  InstallModes
  InstallerSwitches
  InstallerSuccessCodes
  ExpectedReturnCodes
  UpgradeBehavior
  Commands
  Protocols
  FileExtensions
  Dependencies
  Markets
  InstallerAbortsTerminal
  InstallLocationRequired
  RequireExplicitUpgrade
  UnsupportedOSArchitectures
  UnsupportedArguments
  AppsAndFeaturesEntries
  ElevationRequirement
  DisplayInstallWarnings
> Installers
  ReleaseDateTime
  DISPLAY PREVIEW
  SAVE AND EXIT

 Which installer would you like to edit?:
  ALL INSTALLERS
  INSTALLER 1: x64 | wix | machine | en-US | https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x64.msi
  INSTALLER 2: x86 | wix | machine | en-US | https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-5.2.2-x86.msi
  INSTALLER 3: x86 | nullsoft | user | https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe
  INSTALLER 4: x86 | nullsoft | user | https://releases.mattermost.com/desktop/5.2.2/mattermost-desktop-setup-5.2.2-win.exe
  DISPLAY PREVIEW
  SAVE AND EXIT
  ```

**Comment by Trenly** [comment]
> Wingetcreate then generates a `Duplicate installer entry` error and when displaying the manifest, the 3rd is incorrectly listed as `x86` not `x64` as submitted:

I see, thanks for the clarification!

**Comment by Jaifroid** [comment]
I think this is a similar problem to #291. The problem is that wingetcreate used to allow two different entries for the same installer, for example for different architectures or for different scopes (per user, per machine), but at some point an update started to disallow this. There is now no way to create a manifest with two different scopes that use the same installer with this tool. For me, wingetcreate has become completely useless for CI, which is a shame. #291 has been open since July, with no movement, and the sad thing is that it's probably a quite minor change needed to fix this regression (it used to work fine with earlier versions of wingetcreate).

**Comment by mdanish-kh** [comment]
Resolved in [WinGet-Create v1.2.6.0](https://github.com/microsoft/winget-create/releases/tag/v1.2.6.0)

The following command now works to update the example package used in the issue

```
wingetcreate update Mattermost.MattermostDesktop --version 5.5.1 --urls "https://releases.mattermost.com/desktop/5.3.0/mattermost-desktop-5.3.0-x64.msi|x64|machine" "https://releases.mattermost.com/desktop/5.3.0/mattermost-desktop-5.3.0-x86.msi|x86|machine" "https://releases.mattermost.com/desktop/5.3.0/mattermost-desktop-setup-5.3.0-win.exe|x64|user" "https://releases.mattermost.com/desktop/5.3.0/mattermost-desktop-setup-5.3.0-win.exe|x86|user"
```
(Used a dummy version and old installer Urls to simulate an upgrade scenario)

cc @denelon / @ryfu-msft if this can be closed as completed
